### ISSUE TITLE
Significant bottleneck in TFT model training & inference

### ISSUE THREAD

**Issue by daidahao** [unit8co/darts#2897]
**Is your feature request related to a current problem? Please describe.**
There is a significant time bottleneck in the current implementation of TFT. The bottleneck is most significant when the training set is large (>10K samples). On NVIDIA V100 GPU, we observed 90% overheads from `_TimeDistributedInterpolation`. That module is responsible for resampling on embedding dimension in `_ResampleNorm` and `_AddNorm` when the input size does not match the output size. However, we observed that resampling can be skipped or replaced by linear projection without affecting the accuracy of the TFT. 

On our interval datasets, skipping the resampling operation leads to 10x speedup on V100 GPU and 30x speedup on H100 GPU.

**Describe proposed solution**
Provide a `skip_resampling` option for TFT. When set to `True`, the behaviours of TFT are changed as follows:
- Resampling in `flattened_grn` of `_VariableSelectionNetwork` is skipped.
- `_ResampleNorm` for `single_variable_grns` of `_VariableSelectionNetwork` is replaced by a `_ProjectNorm`.
    - `_ProjectNorm` replaces the resampling in `_ResampleNorm` with linear projection.
- Resampling in `_GatedResidualNetwork` for `single_variable_grns` of `_VariableSelectionNetwork` is skipped.

By default, `skip_resampling` is set to `False` and the old behaviours (sampling) of TFT are retained.

**Describe potential alternatives**
Skipping the resampling operation by default, but that might lead to compatibility issues with previously trained TFT checkpoints.

**Additional context**
A Python script for running the TFT training & inference is provided.
```python
import logging
import time

import numpy as np
import pandas as pd

from darts import TimeSeries
from darts.models import TFTModel

logging.basicConfig(level=logging.INFO)

# Generate the dataset and preprocess it
def generate_series(length: int, num_covariates: int):
    times = pd.date_range("20130101", periods=length, freq="s")
    series = TimeSeries.from_times_and_values(
        times=times,
        values=np.cos(np.linspace(0, np.pi * 10, length, dtype=np.float32)).reshape(-1, 1)
    )
    future_covariates = TimeSeries.from_times_and_values(
        times=times,
        values=np.random.rand(length, num_covariates).astype(np.float32)
    )
    return series, future_covariates

series, future_covariates = generate_series(length=10000, num_covariates=15)

# Split the dataset into training and validation sets
train_series, val_series = series.split_after(0.8)

model_kwargs = {
    "batch_size": 512,
    "n_epochs": 1,
    "pl_trainer_kwargs": {
        "accelerator": "gpu",
    },
    "optimizer_kwargs": {"lr": 1e-2},
}
fit_kwargs = {
    "dataloader_kwargs": {
        "num_workers": 0,
    }
}

# Define the TFT model
model = TFTModel(
    input_chunk_length=100,
    output_chunk_length=100,
    # skip_resampling=False,
    **model_kwargs
)

# Train the TFT model
start_time = time.time()
model.fit(
    train_series,
    future_covariates=future_covariates,
    val_series=val_series,
    val_future_covariates=future_covariates,
    verbose=True,
    **fit_kwargs
)
print(f"Training time: {time.time() - start_time:.4f} seconds")
```
